gwcs/tests/test_wcs.py

```diff
@@ -1316,3 +1303,32 @@ def test_spatial_spectral_stokes():
     assert_allclose(gw_sky.data.lat, aw_sky.data.lat)
     assert_allclose(gw_spec.value, aw_spec.value)
     assert_allclose(gw_stokes.value, aw_stokes.value)
+
+
+def test_split_frame_wcs():
+    # Setup a model where the pixel & world axes are (lat, wave, lon)
+    spatial = models.Multiply(10*u.arcsec/u.pix) & models.Multiply(15*u.arcsec/u.pix)  # pretend this is a spatial model
+    compound = models.Linear1D(intercept=0*u.nm, slope=10*u.nm/u.pix) & spatial
+    forward = models.Mapping((1, 2, 0)) | compound | models.Mapping((2, 0, 1))
+
+    # Setup the output frame
+    celestial_frame = cf.CelestialFrame(axes_order=(2, 0), unit=(u.arcsec, u.arcsec),
+                                        reference_frame=coord.ICRS())
+    spectral_frame = cf.SpectralFrame(axes_order=(1,), unit=u.nm)
+    output_frame = cf.CompositeFrame([spectral_frame, celestial_frame])
+
+    input_frame = cf.CoordinateFrame(3, ["PIXEL"]*3,
+                                     axes_order=list(range(3)), unit=[u.pix]*3)
+
+    iwcs = wcs.WCS(forward, input_frame, output_frame)
+    input_pixel = [0*u.pix, 1*u.pix, 2*u.pix]
+    output_world = iwcs.pixel_to_world_values(*input_pixel)
+    output_pixel = iwcs.world_to_pixel_values(*output_world)
+    assert_allclose(output_pixel, u.Quantity(input_pixel).to_value(u.pix))
+
+    world_obj = iwcs.pixel_to_world(*input_pixel)
+    assert isinstance(world_obj[0], coord.SkyCoord)
+    assert isinstance(world_obj[1], coord.SpectralCoord)
```

Comment:
I need to assert if this is the same order as `astropy.wcs` would do it, and just if it seems right.